src/libs/components/back/Back.tsx

```diff
@@ -0,0 +1,29 @@
+import { Link, useLocation } from 'react-router-dom';
+import classNames from 'classnames';
+
+import './styles.scss';
+
+import { Icon } from '../icon/Icon';
+import { IconNames } from '../../enums';
+
+type Props = {
+  className?: string;
+};
+
+export const Back: React.FC<Props> = ({ className }) => {
+  const { state, pathname } = useLocation();
+
+  const backPath = state ? `${state.pathname}${state.search || ''}` : '..';
```

Comment:
**Perhaps** you can use simple button (instead Link) and on `onClick` use `navigate(-1)`: https://reactrouter.com/en/main/hooks/use-navigate

Reply:
I wanted the user to be able to return to /phones with saved search params (page, sort, etc.), and also the user to be able to go to the main page if he came from another domain, so I used the state. 
Redirecting on phones page when user will choose different colours and then click "back" makes sense.